test/tf_exec.cpp

```diff
@@ -360,15 +379,110 @@ TEST(tf_exec, Tile) {
+  std::vector<tf::int64> output_del_size_valid = {1, 3, 2, 2};
+  std::vector<tf::int64> output_del_size_same = {1, 4, 3, 2};
+  tf::Tensor output_delta_valid(tf::DT_FLOAT,
+                                tf::TensorShape(output_del_size_valid));
+  tf::Tensor output_delta_same(tf::DT_FLOAT,
+                               tf::TensorShape(output_del_size_same));
+  AssignInputValues(output_delta_valid);
+  AssignInputValues(output_delta_same);
+
+  std::map<std::string, tf::Tensor*> out_delta_size_map = {
+      {"VALID", &output_delta_valid}, {"SAME", &output_delta_same}};
+
+  std::vector<int> stride = {1, 2, 2, 1};
+  tf::Tensor input_data(tf::DT_FLOAT, tf::TensorShape(input_size_NHWC));
+  AssignInputValues(input_data);
+
+  auto filter_sizes = tf::ops::Const(root, {3, 3, 2, 2});
+
+  tf::ClientSession session(root);
+  std::vector<tf::Tensor> outputs_ngraph;
+  std::vector<tf::Tensor> outputs_cpu;
+
+  // TEST NHWC : default data format
+  for (auto map_iterator : out_delta_size_map) {
+    auto padding_type = map_iterator.first;
+    auto output_delta = *(out_delta_size_map[padding_type]);
+
+    auto r_ngraph = tf::ops::Conv2DBackpropFilter(
+        root_ngraph.WithOpName("r_NGRAPH"), input_data, filter_sizes,
+        output_delta, stride, padding_type);
+
+    auto r_cpu = tf::ops::Conv2DBackpropFilter(
+        root.WithOpName("r_CPU"), input_data, filter_sizes, output_delta,
+        stride, padding_type);
+
+    TF_CHECK_OK(session.Run({r_ngraph}, &outputs_ngraph));
+    TF_CHECK_OK(session.Run({r_cpu}, &outputs_cpu));
+
+    ASSERT_EQ(outputs_ngraph[0].shape(), outputs_cpu[0].shape());
+    AssertTensorEquals(outputs_ngraph[0], outputs_cpu[0]);
+  }
+
+  // TEST NCHW
+  // Dialtion rates > 1 not supported on CPU
+  // Current testing only with dialtion rate 1
+  tf::ops::Conv2DBackpropFilter::Attrs op_attr_nchw;
+  op_attr_nchw = op_attr_nchw.DataFormat("NCHW");
+  op_attr_nchw = op_attr_nchw.Dilations({1, 1, 1, 1});
+
+  tf::ops::Conv2DBackpropFilter::Attrs op_attr_nhwc;
+  op_attr_nhwc = op_attr_nhwc.DataFormat("NHWC");
+  op_attr_nhwc = op_attr_nhwc.Dilations({1, 1, 1, 1});
+
+  for (auto map_iterator : out_delta_size_map) {
+    auto padding_type = map_iterator.first;
+    auto output_delta = *(out_delta_size_map[padding_type]);
+
+    auto input_data_NCHW = tf::ops::Transpose(root, input_data, {0, 3, 1, 2});
+    auto output_delta_NCHW =
+        tf::ops::Transpose(root, output_delta, {0, 3, 1, 2});
+    auto stride_NCHW(stride);
+    stride_NCHW[1] = stride[3];
+    stride_NCHW[2] = stride[1];
+    stride_NCHW[3] = stride[2];
+
+    auto r_ngraph = tf::ops::Conv2DBackpropFilter(
+        root_ngraph.WithOpName("r_NGRAPH"), input_data_NCHW, filter_sizes,
+        output_delta_NCHW, stride_NCHW, padding_type, op_attr_nchw);
+
+    // CPU supports only NHWC
+    auto r_cpu = tf::ops::Conv2DBackpropFilter(
+        root.WithOpName("r_CPU"), input_data, filter_sizes, output_delta,
+        stride, padding_type, op_attr_nhwc);
```

Comment:
Is the output shape for `NHWC` and `NCHW` the same?

Reply:
Yes, I believe so. 
The output shape is same as filter size/shape which remains the same irrespective of NHWC or NCHW data format

https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/conv2-d-backprop-filter
Returns:
Output: 4-D with shape [filter_height, filter_width, in_channels, out_channels]. Gradient w.r.t. the filter input of the convolution.